src/util/ChartUtils.ts

```diff
@@ -1266,18 +1266,28 @@ export const getBandSizeOfAxis = (
     }
   }
 
-  if (axis && ticks && ticks.length >= 2) {
-    const orderedTicks = _.sortBy(ticks, o => o.coordinate);
-    let bandSize = Infinity;
+  // @ts-expect-error we need to rethink width type
```

Comment:
Instead of doing this maybe we re-think the type sent to this function. `BaseAxisProps` does not include `width` but `XAxis` and YAxis` props do. 

What if this type became a union of `XAxisProps` | `YAxisProps` | `ZAxisProps` and type narrowing can be done from there. That way we can add this with correct TS defs.

Reply:
Fixed